backend/src/config/translated_string.rs

```diff
@@ -54,3 +32,32 @@ impl fmt::Debug for TranslatedString {
         f.debug_map().entries(self.0.iter()).finish()
     }
 }
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+#[serde(try_from = "String")]
+pub(crate) enum LangKey {
+    En,
+    De,
+}
+
+impl TryFrom<String> for LangKey {
+    type Error = Error;
+
+    fn try_from(value: String) -> Result<Self, Self::Error> {
+        match value.as_str() {
+            "en" => Ok(Self::En),
+            "de" => Ok(Self::De),
+            _ => Err(anyhow!("Invalid language. Must be either 'en' or 'de'.")),
+        }
+    }
+}
+
+impl fmt::Display for LangKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            LangKey::En => "en",
+            LangKey::De => "de",
+        };
+        write!(f, "{}", s)
+    }
+}
```

Comment:
I just found out about this nifty trick:

```suggestion
impl fmt::Display for LangKey {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        self.serialize(f)
    }
}
```

You can also change that for `Mode` and `Size`